keymanager-api-common/src/lib.rs

```diff
@@ -32,35 +29,8 @@ pub fn set_trusted_policy_signers(signers: TrustedPolicySigners) -> bool {
     true
 }
 
-const POLICY_SIGN_CONTEXT: &[u8] = b"oasis-core/keymanager: policy";
-
-impl SignedPolicySGX {
-    /// Verify the signatures and return the PolicySGX, if the signatures are correct.
-    pub fn verify(&self) -> Result<PolicySGX, KeyManagerError> {
-        // Verify the signatures.
-        let untrusted_policy_raw = cbor::to_vec(self.policy.clone());
-        let mut signers: HashSet<OasisPublicKey> = HashSet::new();
-        for sig in &self.signatures {
-            let public_key = match sig.public_key {
-                Some(public_key) => public_key,
-                None => return Err(KeyManagerError::PolicyInvalid),
-            };
-
-            sig.signature
-                .verify(&public_key, POLICY_SIGN_CONTEXT, &untrusted_policy_raw)
-                .map_err(|_| KeyManagerError::PolicyInvalidSignature)?;
-
-            signers.insert(public_key);
-        }
-
-        // Ensure that enough valid signatures from trusted signers are present.
-        let trusted_signers = TRUSTED_SIGNERS.lock().unwrap();
-        let signers: HashSet<_> = trusted_signers.signers.intersection(&signers).collect();
-        let multisig_threshold = trusted_signers.threshold;
-        if signers.len() < multisig_threshold as usize {
-            return Err(KeyManagerError::PolicyInsufficientSignatures);
-        }
-
-        Ok(self.policy.clone())
-    }
+/// Verify that policy has valid signatures and that enough of them are from the global set
+/// of trusted policy signers.
+pub fn verify_policy_and_trusted_signers(policy: &SignedPolicySGX) -> Result<(), KeyManagerError> {
```

Comment:
This (and maybe even the `verify` method) could even return the `policy.policy` so upstream callers can do something like `let policy = verify_policy_and_trusted_signers(untrusted_policy)?;` which seems to be common now.

Reply:
I applied your comment. Policy is now returned (just as it was before this PR).

I intentionally wanted to removed this for 4 reasons:
- The `verify` method had two responsibilities, as can be seen from the comment `/// Verify the signatures and return the PolicySGX`.
- I expect that `verify` returns `true/false` or `()/Err`, and not a policy. However, I expect `sign` to return a signed policy.
- Policy had to be cloned, to achieve that `Ok(self.policy.clone())`. Unnecessary step.
- Passing policy back and forth just doesn't feel right as it can always be accessed via `SignedPolicySGX`. It would be nice if this would give us some kind of protection, but it doesn't. For example, if policy was private and could be accessed only via getter function `verify`, then everyone would be forced to call `verify`.

The only reason for is to have more readable code `let policy = verify_policy...`.

Reply:
There should be no need to clone a policy, you should be able to do something like:
```rust
pub fn verify<'a>(policy: &'a SignedPolicySGX) -> Result<&'a PolicySGX, Error> {
```
and then return a reference, e.g. `&policy.policy`.

> For example, if policy was private and could be accessed only via getter function verify, then everyone would be forced to call verify.

Yeah that would be ideal but not sure how it would work with deserialization.